src/util/support/threads.c

```diff
@@ -237,7 +237,7 @@ void *k5_getspecific (k5_key_t keynum)
     if (err)
         return NULL;
 
-    assert(keynum >= 0 && keynum < K5_KEY_MAX);
+    assert(keynum < K5_KEY_MAX);
```

Comment:
Hmm, I wonder why clang warns so aggressively there then.  If you're find removing the assert entirely, I'll go ahead and do that.